iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/receiver/protocol/thrift/IoTDBDataNodeReceiver.java

```diff
@@ -319,6 +328,71 @@ protected String getClusterId() {
     return IoTDBDescriptor.getInstance().getConfig().getClusterId();
   }
 
+  @Override
+  protected TSStatus asyncLoadTsFile(final String fileAbsolutePath) throws Exception {
+    moveToLoadTsFileDirBySequence.set(IOTDB_CONFIG.getLoadActiveListeningPipeDir());
+
+    final File sourceFile = new File(fileAbsolutePath);
+    // whether the move folder equal the listening folder
+    if (moveToLoadTsFileDirBySequence.get().equals(sourceFile.getParentFile().getAbsolutePath())) {
```

Comment:
Will users directly make "load_dir" and "pipe_receiver_dir" the same? That may cause loader to load writing tsFile which is not complete.. Will generate a lot of failure logs and waste some resources.
It's actually OK to allow it but I don't think this shall be an encouraged behaviour...

Reply:
Besides, "pipe_receiver_dir" has one receiver-id suffix, like /receiver/13/xxxx..... It will not directly equal to the loading dir because loading dir has no id.

Reply:
> Besides, "pipe_receiver_dir" has one receiver-id suffix, like /receiver/13/xxxx..... It will not directly equal to the loading dir because loading dir has no id.

The file loading has been added with logic to determine whether the file has been written. If it is not finished, it will not be scanned. Then if there are subdirectories in the configured directory, it will be recursively scanned.